### PR TITLE
Implement upload ordering

### PR DESCRIPTION
**Pull request by villesau**
This pull request introduces new option called `priority`. It takes array of regexes in and prioritises from higher to lower priority, and will upload files in priority order. All the files which are not matched by any regexp gets highest priority and will be uploaded first. This allows us to set priority to any file, and e.g postpone html file uploading until other files have been successfully uploaded. This prevent's users to access assets which are still uploading.

potential configurations:
`priority: [/js/, /css/, /html/]` first js, then css, then html
`priority: [/html/]` uploads html last since rest of the files are not matched by priority array
`priority: [/html/, /.*/]` uploads html first, others after as rest of the files matches the second regexp

**Why this is important?**

If file is requested before uploading, the object enters in eventual consistent mode: It's not available before it's distributed to all the servers. This might take minutes and in the mean time 404 is returned even after successful upload. This might happen if html file is uploaded before assets, and user accesses the website just before other assets are uploaded which triggers loading the rest of the assets while they are still uploading to S3. If this happens, the site might be down for several minutes since s3 returns 404.

further reading about S3 consistency model:
https://docs.aws.amazon.com/AmazonS3/latest/dev/Introduction.html#ConsistencyModel
https://codeburst.io/quick-explanation-of-the-s3-consistency-model-6c9f325e3f82

### CODE REVIEW

File: test/upload_test.js [hunk 1]
```diff
@@ -86,6 +86,31 @@ describe('S3 Webpack Upload', function() {
...
+            const cssFileName = files.find(isCss)
+            const jsFilename = files.find(isJs)
+
+            return Promise.all([
+              testHelpers.getS3Object(cssFileName),
+              testHelpers.getS3Object(jsFilename)
+            ])
+          })
+          .then(([cssObject, jsObject]) => {
+            assert.isTrue(cssObject.LastModified.getTime() >= jsObject.LastModified.getTime())
```
Comment (villesau) [inline test/upload_test.js]: Note that S3 accuracy is 1 second so it's not possible to test that the desired object _actually_ is uploaded later: Files are likely te be uploaded at the same second. This test is to ensure that it's at leat not uploaded before. If you want to see the real behaviour, I'd suggest to add logging to e.g in uploadPriorityChunk method.

**Comment by villesau** [comment]
ping @MikaAK

**Comment by MikaAK** [comment]
Hey @villesau thanks for the PR, I've seen this but I haven't had a chance to go through it's utility, I'll read the two links provided and update this over the weekend

File: test/upload_test_helpers.js [hunk 2]
```diff
@@ -224,5 +225,23 @@ export default {
...
+
+  getS3Object(key) {
+    const s3 = new S3({
+      accessKeyId: s3Opts.AWS_ACCESS_KEY,
+      secretAccessKey: s3Opts.AWS_SECRET_ACCESS_KEY
+    })
+
+
+    return new Promise((resolve, reject) => {
+      s3.getObject({Bucket: s3Opts.AWS_BUCKET, Key: key}, function(err, data) {
```
Comment (capaj) [inline test/upload_test_helpers.js]: @MikaAK you can do `s3.getObject().promise()` and thus avoid having ot create a new promise yourself

**Pull request merged**